Mathlib/Algebra/BigOperators/Fin.lean

```diff
@@ -475,3 +477,21 @@ theorem alternatingProd_eq_finset_prod {G : Type*} [CommGroup G] :
           simp [pow_add]}
 
 end List
+
+namespace Finsupp
+
+variable {M N : Type*}
+
+lemma sum_cons [AddCommMonoid M] (n : ℕ) (σ : Fin n →₀ M) (i : M) :
+    (sum (cons i σ) fun _ e ↦ e) = i + sum σ (fun _ e ↦ e) := by
+  rw [sum_fintype _ _ (fun _ => rfl), sum_fintype _ _ (fun _ => rfl)]
+  exact Fin.sum_cons i σ
+
+lemma sum_cons' [AddCommMonoid M] [AddCommMonoid N] (n : ℕ) (σ : Fin n →₀ M) (i : M)
+    (f : Fin (n+1) → M → N) (h : ∀ x, f x 0 = 0) :
+    (sum (Finsupp.cons i σ) f) = f 0 i + sum σ (Fin.tail f) := by
+  rw [sum_fintype _ _ (fun _ => by apply h), sum_fintype _ _ (fun _ => by apply h)]
+  simp_rw [Fin.sum_univ_succ, cons_zero, cons_succ]
+  congr
+
+end Finsupp
```

Comment:
I'd be inclined to move these to a new `BigOperators/Finsupp/Fin.lean`, which should further reduce the weight on downstream files that don't need `Finsupp`.

Reply:
Otherwise,

bors d+

Reply:
It's going to be a rather small file, which is why I started by flipping the import direction would be better. But small files are no issue for me! :)